python-checks/src/test/resources/checks/booleanCheckNotInverted.py

```diff
@@ -0,0 +1,51 @@
+a = 1
+b = 2
+c = not (a == b)  # Noncompliant {{Use the opposite operator ("!=") instead.}}
+#   ^^^^^^^^^^^^
+cc = not (a != b)  # Noncompliant
+
+d = not (a > b)  # Noncompliant
+
+e = not (a <= b)  # Noncompliant
+
+f = float('nan')
+
+g = not (f > 2)  # Noncompliant
+
+h = not (f >= 2)  # Noncompliant
+
+i = not (((((a * 2)))))
+
+j = a and (not b)
+
+k = (not a) and (not b)
+
+l = not ((((((a > b))))))  # Noncompliant
+
+o = not a == b  # Noncompliant
+
+m = not a == b == 1
+
+n = not (not (a == b))  # Noncompliant
+
+p = not (1 == a == b) == 2 # Noncompliant
+
+q = not (a == 1) == b == c == (d and e)
+
+r = not (a and 1) == b == c == (d and e)
+s = a is not b
+
+t = not (a is b)  # Noncompliant {{Use the opposite operator ("is not") instead.}}
+#   ^^^^^^^^^^^^
+u = not (a is not b)  # Noncompliant
```

Comment:
I believe that adding an assertion on the issue message could be valuable, at least to check it's not "is not not".